src/components/interaction-designer/block-types/Core_SetGroupMembershipBlock.vue

```diff
@@ -0,0 +1,125 @@
+                         :options="actionsList"
+                         :allow-empty="true"
+                         :show-labels="false"
+                         :searchable="false">
+        </vue-multiselect>
+      </div>
+
+      <group-selector :block="block"/>
+
+      <first-block-editor-button
+        :flow="flow"
+        :block-id="block.uuid"/>
+
+    </fieldset>
+
+    <block-id :block="block"/>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import { namespace } from 'vuex-class'
+import { Component, Prop } from 'vue-property-decorator'
+
+import {
+  IFlow,
+  IBlock,
+  ISetGroupMembershipBlockConfig,
+} from '@floip/flow-runner'
+import BlockNameEditor from '../block-editors/NameEditor.vue'
+import BlockLabelEditor from '../block-editors/LabelEditor.vue'
+import BlockSemanticLabelEditor from '../block-editors/SemanticLabelEditor.vue'
+import FirstBlockEditorButton from '../flow-editors/FirstBlockEditorButton.vue'
+import BlockId from '../block-editors/BlockId.vue'
+import GroupSelector from '@/components/interaction-designer/block-editors/GroupSelector.vue'
+import VueMultiselect from 'vue-multiselect'
+
+import SetGroupMembershipStore, { BLOCK_TYPE, ADD_KEY, REMOVE_KEY } from '@/store/flow/block-types/Core_SetGroupMembershipStore'
+import lang, { trans } from '@/lib/filters/lang'
+import { createDefaultBlockTypeInstallerFor } from '@/store/builder'
+import { find } from 'lodash'
+
+const blockVuexNamespace = namespace(`flow/${BLOCK_TYPE}`)
+const flowVuexNamespace = namespace('flow')
+const builderVuexNamespace = namespace('builder')
+
+interface IGroupOption {
+  id: string;
+  name: string;
+}
+
+//providing this generic is required by tsserver checking but not in the build run by yarn storybook
+//TODO - understand what is going on here and if there is something more correct we should have instead
+@Component<any>({
+  components: {
+    BlockNameEditor,
+    BlockLabelEditor,
+    BlockSemanticLabelEditor,
+    FirstBlockEditorButton,
+    BlockId,
+    GroupSelector,
+    VueMultiselect,
+  },
+  mixins: [lang],
+})
+class Core_SetGroupMembershipBlock extends Vue {
+  @Prop() readonly block!: IBlock
+  @Prop() readonly flow!: IFlow
+
+  actionsList: IGroupOption[] = [
+    {
+      id: ADD_KEY,
+      name: trans('flow-builder.add'),
+    },
+    {
+      id: REMOVE_KEY,
+      name: trans('flow-builder.remove'),
+    },
+  ]
+
+  get selectedAction() {
+    const { isMember } = this.block.config as ISetGroupMembershipBlockConfig
+    //TODO: we can remove the safe cast JSON.parse(isMember) once ISetGroupMembershipBlockConfig isMember type is changed to boolean
+    if (JSON.parse(isMember) === false) {
+      return find(this.actionsList, { id: REMOVE_KEY }) || null
+    }
+
+    if (JSON.parse(isMember) === true) {
+      return find(this.actionsList, { id: ADD_KEY }) || null
+    }
+
+    return null
+  }
+
+  set selectedAction(group: IGroupOption) {
+    this.setIsMemberFromGroup(group)
```

Comment:
I'm not expecting any changes here, but wanted to take this opportunity to share some food for thought: In the future, can we try to consolidate where "behaviour" is encapsulated? 

In this case we have some special domain constraints that has isMember correlate directly to an action over a groupId, except we have half of the behaviour in vuex and half of it here as the getter. From my perspective, it could be nice to encapsulate both counterparts as part of this computed property so that we don't need to leak UI layer's correlation onto many layers. 

So the vuex action would end up being dirt simple, like: `setIsMember(true | false)` and the translation happens at the same layer for both get and set of action.

Reply:
Thanks for explaining, noting this ;)